### ISSUE TITLE
Codecov reports coverage changes in seemingly random files

### ISSUE THREAD

**Issue by kevinburke** [sourcegraph/sourcegraph#319]
For example, here, https://github.com/sourcegraph/sourcegraph/pull/311#issuecomment-427526349

that PR changed the dev scripts, but Codecov reported that some '*.go' files changed coverage which doesn't make sense.

**Comment by felixfbecker** [comment]
I don't see a coverage report on that PR?

I would trust that codecov accurately reports which lines were executed during tests. But if there is code that is undeterministacally sometimes hit during tests and sometimes not (e.g. timed background tasks) then the coverage becomes non-deterministic.

**Comment by keegancsmith** [comment]
Yeah I have been finding this annoying. Generally it has to do with a background goroutine racing with the end of the test. It certainly helps with training you to ignore the codecov PR comments, so some time should be spent to fix the worst offenders. The annoying part is when you try click on an unrelated file, codecov doesn't take you to see the file since it wasn't modified in the PR.

**Comment by kevinburke** [comment]
Here's one. https://github.com/sourcegraph/sourcegraph/pull/309
